### PR TITLE
{toolchain} intel/2019a (+ HPL 2.3 as test case)

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node3102.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/6cd93ecb41117c6e4552978239d7968b for a full test report.

**Comment by hajgato** [comment]
@boegel change foss to intel in the title?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2094.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/8da6a5e40a67fdc97475bd45e0cf63d0 for a full test report.

**Comment by boegel** [comment]
@hajgato Woops, fixed (incl. commit message with `git push --force`)

**Comment by damianam** [comment]
`Intel MPI 2019.[0,1]` crashes when using most collectives, for most algorithms. That's why I think we should use `Intel MPI 2018.4` instead, unless someone finds the underlying problem with the new versions (potentially related to `libfabric`?)

**Comment by akesandgren** [comment]
IMPI 2019.1 fails on IMB 4.1 Reduce with 1024 tasks (4 bytes) for me, so definitely some problem with it.

**Comment by akesandgren** [comment]
And IMPI 2018.3.222 (intel/2018b) works as it should.

So clearly we need to stay away from IMPI 2019.[01]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node3111.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/f0e5d0de69f19c722eb54f39cc4ed72e for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2617.swalot.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/97454a9f8d0c93719e42a4bd93adb6a3 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2001.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/0c9cfedb5c539fda8c653d35c72edcac for a full test report.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
grc-cluster1 - Linux centos 6.10, Intel(R) Xeon(R) CPU E5-2640 0 @ 2.50GHz, Python 2.7.14
See https://gist.github.com/afda356cf5d5dfc0363eb251bc39d2ce for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 3 out of 5 (5 easyconfigs in this PR)
b-cn0315.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/c8c6cb5d66fcbc2853f9d5f70a7f7cb8 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
b-cn0315.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/9e604fa098778d39011878742f2b9243 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 8 (5 easyconfigs in this PR)
nic27 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/1d73ff8ff2dffa9da744451ab242c9a2 for a full test report.

**Comment by boegel** [comment]
Detailed test report on re-building all easyconfigs that use `intel/2018b` as toolchain with the proposed `intel/2019a` instead is available at https://gist.github.com/boegel/68b542fa564fd370e8b0e7710a320f6f .

Compilation failed for a handful of easyconfigs with cryptic errors (to be figured out), but I don't think any of these should block us from going forward with the current `intel/2019a` definition (and this was not contested during the last EasyBuild conf call, see https://github.com/easybuilders/easybuild/wiki/Conference-call-notes-20190123#test-report-for-proposed-2019a-common-toolchains).

Summary: 110 successfully installed easyconfigs using `intel/2019a` (+ 102 more with `GCCcore/8.2.0` subtoolchain) vs 3+2 failed installations.

File: easybuild/easyconfigs/h/HPL/HPL-2.3-intel-2019a.eb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+source_urls = ['http://www.netlib.org/benchmark/%(namelower)s']
+sources = [SOURCELOWER_TAR_GZ]
+patches = ['HPL_parallel-make.patch']
+checksums = [
+    '32c5c17d22330e6f2337b681aded51637fb6008d3f0eb7c277b163fadd612830',  # hpl-2.3.tar.gz
+    '2a5bf9c4f328049828ddecec7ba3f05a9e25d236f4212747c53bd22fea80c5e6',  # HPL_parallel-make.patch
+]
+
+# fix Make dependencies, so parallel build also works
```
Comment (ocaisa) [inline easybuild/easyconfigs/h/HPL/HPL-2.3-intel-2019a.eb]: Move comment to appropriate place

File: easybuild/easyconfigs/i/iimpi/iimpi-2019a.eb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# This is an easyconfig file for EasyBuild, see http://easybuilders.github.io/easybuild
```
Comment (ocaisa) [inline easybuild/easyconfigs/i/iimpi/iimpi-2019a.eb]: Not sure we should bother including this comment

File: easybuild/easyconfigs/i/imkl/imkl-2019.1.144-iimpi-2019a.eb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# This is an easyconfig file for EasyBuild, see https://easybuilders.github.io/easybuild/
```
Comment (ocaisa) [inline easybuild/easyconfigs/i/imkl/imkl-2019.1.144-iimpi-2019a.eb]: Ditto

File: easybuild/easyconfigs/i/impi/impi-2018.4.274-iccifort-2019.1.144-GCC-8.2.0-2.31.1.eb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# This is an easyconfig file for EasyBuild, see http://easybuilders.github.io/easybuild
```
Comment (ocaisa) [inline easybuild/easyconfigs/i/impi/impi-2018.4.274-iccifort-2019.1.144-GCC-8.2.0-2.31.1.eb]: Ditto

File: easybuild/easyconfigs/i/intel/intel-2019a.eb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+easyblock = 'Toolchain'
+
+name = 'intel'
+version = '2019a'
+
+homepage = 'http://software.intel.com/en-us/intel-cluster-toolkit-compiler/'
```
Comment (ocaisa) [inline easybuild/easyconfigs/i/intel/intel-2019a.eb]: Wonder if we should change this to https://easybuild.readthedocs.io/en/develop/Common-toolchains.html#intel-toolchain

**Comment by jordiblasco** [comment]
I'm not sure if you have seen that there is a new release of Intel MPI (2019U2)
https://software.intel.com/en-us/articles/intel-mpi-library-release-notes-linux#inpage-nav-3-1

**Comment by boegel** [comment]
@jordiblasco see #7598

**Pull request merged**